### PR TITLE
[21808] Update DataWriter/Reader `get_matched_publication/subscription...()` (2.x) API tests

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR updates DataWriter/Reader `get_matched_publication/subscirption...()` tests in the `Fast DDS Python 2.x branch` as result of API implementation:

* eProsima/Fast-DDS#5312

**IMPORTANT**
This PR must be merged after 
* #193 

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally
- **N/A** Changes are API compatible. 
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by Mario-DL** [comment]
IMO This PR should target a `2.1.x` branch

**Comment by Mario-DL** [comment]
IMO this PR should target the minor attached to `Fast DDS 3.2.x`

File: fastdds_python/test/api/test_datareader.py [hunk 1]
```diff
@@ -773,25 +773,35 @@ def test_get_liveliness_changed_status(datareader):
...
     - DataReader::get_matched_publications
     """
     ihs = fastdds.InstanceHandleVector()
-    assert(fastdds.RETCODE_UNSUPPORTED ==
+    time.sleep(1)
+    assert(fastdds.RETCODE_OK ==
            datareader.get_matched_publications(ihs))
+    # The datawriter (added in the fixture) is the only
+    # one that should be matched
+    assert(1 == ihs.size())
```
Comment (richiware) [inline fastdds_python/test/api/test_datareader.py:804]: Check the instance_handle in the vector is the same that the datawriter.

File: fastdds_python/test/api/test_datawriter.py [hunk 2]
```diff
@@ -308,26 +308,40 @@ def test_get_liveliness_lost_status(datawriter):
     assert(0 == status.total_count_change)
 
 
-def test_get_matched_subscription_data(datawriter):
+def test_get_matched_subscription_data(datawriter, reader_topic, subscriber):
```
Comment (richiware) [inline fastdds_python/test/api/test_datawriter.py:311]: why don't create a pytest fixture `datareader` that could be used in future tests?

**Review by richiware — CHANGES_REQUESTED** [review]

**Review by richiware — APPROVED** [review]

**Pull request merged**